examples/multisig_plain/lib.rs

```diff
@@ -0,0 +1,231 @@
+            self.is_owner.remove(&owner);
+            self.required.set(required);
+            self.clean_owner_confirmations(&owner);
+        }
+
+        #[ink(message)]
+        fn replace_owner(&mut self, owner: AccountId, new_owner: AccountId) {
+            self.ensure_from_wallet();
+            self.ensure_owner(&owner);
+            self.ensure_no_owner(&new_owner);
+            self.owners.replace(self.owner_index(&owner), || new_owner);
+            self.is_owner.remove(&owner);
+            self.is_owner.insert(new_owner, ());
+            self.clean_owner_confirmations(&owner);
+        }
+
+        #[ink(message)]
+        fn change_requirement(&mut self, requirement: u32) {
+            self.ensure_from_wallet();
+            ensure_requirement(self.owners.len(), requirement);
+            self.required.set(requirement);
+        }
+
+        #[ink(message)]
+        fn submit_transaction(&mut self, transaction: Transaction) {
+            self.ensure_from_owner();
+            let id = self.transactions.put(transaction);
+            self.confirmation_count.insert(id, 0);
+            self.internal_confirm(id);
+        }
+
+        #[ink(message)]
+        fn cancel_transaction(&mut self, id: TransactionId) {
+            self.ensure_from_wallet();
+            self.take_transaction(id);
+        }
+
+        #[ink(message)]
+        fn confirm_transaction(&mut self, id: TransactionId) {
+            self.ensure_from_owner();
+            self.ensure_transaction_exists(id);
+            self.internal_confirm(id);
+        }
+
+        #[ink(message)]
+        fn revoke_confirmation(&mut self, id: TransactionId) {
+            self.ensure_from_owner();
+            if self
+                .confirmations
+                .remove(&(id, self.env().caller()))
+                .is_some()
+            {
+                self.confirmation_count
+                    .mutate_with(&id, |count| *count -= 1);
+            }
+        }
+
+        #[ink(message)]
+        fn execute_transaction(&mut self, id: TransactionId) -> Result<(), ()> {
+            self.ensure_confirmed(id);
+            let t = self.take_transaction(id).unwrap();
+            env::call::CallParams::<MyEnv, ()>::invoke(t.callee, t.selector.into())
+                .gas_limit(t.gas_limit)
+                .transferred_value(t.transferred_value)
+                .push_arg(&CallInput(&t.input))
+                .fire()
+                .map(|_| ())
+                .map_err(|_| ())
+        }
+
+        fn internal_confirm(&mut self, id: TransactionId) {
+            if self
+                .confirmations
+                .insert((id, self.env().caller()), ())
+                .is_none()
+            {
+                self.confirmation_count
+                    .mutate_with(&id, |count| *count += 1);
+            }
+        }
+
+        fn owner_index(&self, owner: &AccountId) -> u32 {
+            self.owners.iter().position(|x| *x == *owner).unwrap() as u32
```

Comment:
use `expect` instead of `unwrap`

Reply:
might it makes sense for `self.owners` to be of type `storage::HashMap` or `storage::BTreeMap` instead? Why is it a `storage::Vec`?

Reply:
Because I have to iterate over them in `clean_transaction_confirmations`. As matter of fact there is also a owner HashMap called `is_owner`. This is used to avoid iteration on the more common operations.

Reply:
still requires change to `expect` with proper message

Reply:
It is. This snippet is just outdated.